pkg/apis/cr/v1alpha1/types.go

```diff
@@ -307,3 +308,93 @@ type ProfileList struct {
 	metav1.ListMeta `json:"metadata"`
 	Items           []*Profile `json:"items"`
 }
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +kubebuilder:subresource:status
+
+// RepositoryServer manages the lifecycle of Kopia Repository Server within a Pod
+type RepositoryServer struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+	Spec              RepositoryServerSpec   `json:"spec"`
+	Status            RepositoryServerStatus `json:"status"`
+}
+
+// RepositoryServerSpec is the specification for the RepositoryServer
+type RepositoryServerSpec struct {
+	Storage    Storage    `json:"storage,omitempty"`
+	Repository Repository `json:"repository,omitempty"`
+	Server     Server     `json:"server,omitempty"`
+}
+
+// Storage references the backend store where a repository already exists
+// and the credential necessary to connect to the backend store
+type Storage struct {
+	Secret           corev1.SecretReference `json:"secret,omitempty"`
```

Comment:
Should we rename it to LocationSecret?

Reply:
Right now, `Location` keyword is closely associated with the Profile CR so I avoided it in the RepositoryServer CR.
Are these fields confusing to read as `.spec.storage.secret` and `.spec.storage.credentialSecret`?